jsonrpcbase/main.py

```diff
@@ -289,14 +245,11 @@ def _get_id(self, rdata):
         InvalidRequestError will be raised if the id value has invalid type.
         """
         if 'id' in rdata:
-            if isinstance(rdata['id'], six.string_types) or \
-                    isinstance(rdata['id'], six.integer_types) or \
-                    isinstance(rdata['id'], float) or \
-                    rdata['id'] is None:
+            if type(rdata['id']) in (str, int, float):
```

Comment:
From [JSON RPC 2.0 spec][spec]:

> id
>
> An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null [1] and Numbers SHOULD NOT contain fractional parts [2]

Let's do a follow-up to fix this, most likely with JSON schema.

[spec]: https://www.jsonrpc.org/specification#request_object